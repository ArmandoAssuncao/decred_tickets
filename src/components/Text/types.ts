export interface IProps {
    children?: string
    id?: string
    m: string
}

export interface IState {
}
