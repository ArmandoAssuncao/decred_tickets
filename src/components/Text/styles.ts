import { TextStyle } from 'react-native'
import EStyleSheet from 'react-native-extended-stylesheet'

export default EStyleSheet.create({
    text: {
        color: '$textColor',
    } as TextStyle,
})
