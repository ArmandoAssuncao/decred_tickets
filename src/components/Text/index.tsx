import React from 'react'
import { Text as RNText } from 'react-native'

import { IProps, IState } from './types'
import styles from './styles'

import i18n from 'config/i18n'

export default class Text extends React.Component<IProps, IState> {
    public static defaultProps: Partial<IProps> = {
    }

    constructor(props) {
        super(props)
    }

    render(): JSX.Element {
        const { children, id, m } = this.props
        return id
            ? <RNText style={styles.text}>{i18n.t(id, { defaultValue: m })}</RNText>
            : <RNText style={styles.text}>{children}</RNText>
    }
}
