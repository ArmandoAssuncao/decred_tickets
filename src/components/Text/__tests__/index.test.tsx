import React from 'react'
import 'react-native'
import { shallow } from 'enzyme'

import Text from '../index'

describe('Text', () => {
    it('match snapshot', () => {
        const wrapper = shallow(<Text id={'testId'} m="default value">text</Text>)
        expect(wrapper).toMatchSnapshot()
    })

    it('has text component', () => {
        const wrapper = shallow(<Text m="default value">text</Text>)
        expect(wrapper.find('Text')).toHaveLength(1)
    })
})
