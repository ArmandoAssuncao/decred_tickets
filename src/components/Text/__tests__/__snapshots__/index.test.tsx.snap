// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text match snapshot 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "white",
    }
  }
>
  default value
</Text>
`;
